### PR TITLE
Fix stuck state from period offline

### PR DESCRIPTION
**Pull request by benaadams**
## Changes

- Fix if node is offline too long it ceases to be able to sync properly if pivots move

#### What types of changes does your code introduce?

- [x] Bugfix (a non-breaking change that fixes an issue)
- [ ] New feature (a non-breaking change that adds functionality)
- [ ] Breaking change (a change that causes existing functionality not to work as expected)
- [ ] Optimization
- [ ] Refactoring
- [ ] Documentation update
- [ ] Build-related changes
- [ ] Other: _Description_

#### Requires testing

- [ ] Yes
- [x] No

#### If yes, did you write tests?

- [ ] Yes
- [x] No

### CODE REVIEW

File: src/Nethermind/Nethermind.Synchronization/ParallelSync/MultiSyncModeSelector.cs [hunk 1]
```diff
@@ -590,8 +590,8 @@ private bool ShouldBeInStateSyncMode(Snapshot best)
             bool notInFastSync = !best.IsInFastSync;
             bool notNeedToWaitForHeaders = NotNeedToWaitForHeaders;
             bool stickyStateNodes = best.TargetBlock - best.Header < (FastSyncLag + StickyStateNodesDelta);
-            bool stateNotDownloadedYet = (best.TargetBlock - best.State > FastSyncLag ||
-                                          best.Header > best.State && best.Header > best.Block);
+            bool stateNotDownloadedYet = best.State <= 0 ||
+                                         (best.Header > best.State && best.Header > best.Block);
```
Comment (LukaszRozmej) [inline src/Nethermind/Nethermind.Synchronization/ParallelSync/MultiSyncModeSelector.cs:594]: So is the problem that `FastSyncLag` wasn't working properly? Or maybe the actual problem was the parenthesis were wrong? I expect the latter.

Comment (benaadams) [reply src/Nethermind/Nethermind.Synchronization/ParallelSync/MultiSyncModeSelector.cs:594]: FastSyncLag is 32; I missed 47k blocks while offline and in the 6 days following while it was in a limbo state of constantly switching to StateSync. Changing to `best.State <= 0` it was back online and fully synced in 2hrs.

Comment (LukaszRozmej) [reply src/Nethermind/Nethermind.Synchronization/ParallelSync/MultiSyncModeSelector.cs:594]: `FastSyncLag` shouldn't be `32` by any default

Comment (benaadams) [reply src/Nethermind/Nethermind.Synchronization/ParallelSync/MultiSyncModeSelector.cs:594]: Its a constant, was changed from `2` to `32` over a year ago https://github.com/NethermindEth/nethermind/pull/3944

Comment (benaadams) [reply src/Nethermind/Nethermind.Synchronization/ParallelSync/MultiSyncModeSelector.cs:594]: Issue is if you've been processing blocks; should it ever try to switch to state sync?

Comment (LukaszRozmej) [reply src/Nethermind/Nethermind.Synchronization/ParallelSync/MultiSyncModeSelector.cs:594]: I think what we would expect is `IsInAStickyFullSyncMode` to be true, wonder why it wasn't

Comment (benaadams) [reply src/Nethermind/Nethermind.Synchronization/ParallelSync/MultiSyncModeSelector.cs:594]: Think same scenario @MarekM25 was looking at (different area) maybe?
```csharp
private bool IsInAStickyFullSyncMode(Snapshot best)
{
    long bestBlock = Math.Max(best.Processed, LastBlockThatEnabledFullSync ?? 0);
    bool hasEverBeenInFullSync = bestBlock > _pivotNumber && best.State > _pivotNumber;
    long heightDelta = best.TargetBlock - bestBlock;
    return hasEverBeenInFullSync && heightDelta < FastSyncCatchUpHeightDelta;
}
```
Pivot update (new version or get latest); but head is behind the new pivot?

Comment (LukaszRozmej) [reply src/Nethermind/Nethermind.Synchronization/ParallelSync/MultiSyncModeSelector.cs:594]: Hm... Maybe then just change:

`bool hasEverBeenInFullSync = bestBlock > _pivotNumber && best.State > _pivotNumber;`

to

`bool hasEverBeenInFullSync = bestBlock > 0 && best.State > 0; ?

File: src/Nethermind/Nethermind.Synchronization/ParallelSync/MultiSyncModeSelector.cs [hunk 2]
```diff
@@ -298,7 +298,7 @@ private void UpdateSyncModes(SyncMode newModes, string? reason = null)
         private bool IsInAStickyFullSyncMode(Snapshot best)
         {
             long bestBlock = Math.Max(best.Processed, LastBlockThatEnabledFullSync ?? 0);
-            bool hasEverBeenInFullSync = bestBlock > _pivotNumber && best.State > _pivotNumber;
```
Comment (benaadams) [inline src/Nethermind/Nethermind.Synchronization/ParallelSync/MultiSyncModeSelector.cs:301]: pivots move and can move in front of bestblock etc

Comment (MarekM25) [reply src/Nethermind/Nethermind.Synchronization/ParallelSync/MultiSyncModeSelector.cs:301]: I think this change makes sense. I'm wondering if we can write unit tests to cover it

Comment (benaadams) [reply src/Nethermind/Nethermind.Synchronization/ParallelSync/MultiSyncModeSelector.cs:301]: Existing tests cover it? But don't cover moving pivots?

Comment (MarekM25) [reply src/Nethermind/Nethermind.Synchronization/ParallelSync/MultiSyncModeSelector.cs:301]: What I mean write a test that is failing for the previous version of code and it passes after fix

Comment (benaadams) [reply src/Nethermind/Nethermind.Synchronization/ParallelSync/MultiSyncModeSelector.cs:301]: Made issue for test improvement https://github.com/NethermindEth/nethermind/issues/5734

**Review by LukaszRozmej — APPROVED** [review]

**Pull request merged**